creator-node/src/diskManager.js

```diff
@@ -187,6 +195,145 @@ class DiskManager {
 
     return ret
   }
+
+  static async deleteFileOrDir(pathToFileOrDir) {
+    // Base case - delete single file (not a directory)
+    if (!(await fs.lstat(pathToFileOrDir)).isDirectory()) {
+      await fs.unlink(pathToFileOrDir)
+      return
+    }
+
+    // Recursively remove all contents of directory
+    for (const file of await fs.readdir(pathToFileOrDir)) {
+      const childPath = path.join(pathToFileOrDir, file)
+      if ((await fs.lstat(childPath)).isDirectory()) {
+        await DiskManager.deleteFileOrDir(childPath)
+      } else {
+        await fs.unlink(childPath)
+      }
+    }
+
+    // Remove actual directory
+    await fs.rmdir(pathToFileOrDir)
+  }
+
+  /**
+   * Recursively deletes an array of file paths and their subdirectories in paginated chunks.
+   * @param {string[]} storagePaths the file paths to delete
+   * @param {number} batchSize the number of concurrent deletes to perform
+   * @param {bunyan.Logger} logger
+   * @returns {number} number of files successfully deleted
+   */
+  static async batchDeleteFileOrDir(storagePaths, batchSize, logger) {
+    let numFilesDeleted = 0
+    const batches = chunk(storagePaths, batchSize)
+    const promiseResults = []
+    for (const batchOfStoragePaths of batches) {
+      const promiseResultsForBatch = await Promise.allSettled(
+        batchOfStoragePaths.map((storagePath) =>
+          DiskManager.deleteFileOrDir(storagePath)
+        )
+      )
+      promiseResults.push(...promiseResultsForBatch)
+    }
+
+    // Count number of files successfully deleted and log errors
+    for (const promiseResult of promiseResults) {
+      if (promiseResult.status === 'fulfilled') {
+        numFilesDeleted++
```

Comment:
makes sense this is best effort since files could fail to be deleted if they just don't exist for any reason

Reply:
yeah exactly 🙂